src/bramble/client/ProjectLimiter.js

```diff
@@ -0,0 +1,94 @@
+define(function() {
+"use strict";
+    var ProjectLimiter = {};
+
+    var maxProjectState = {
+            fileSize: 0, // max size of any file in bytes. NOT USED now.
+            fileCount: 0, // total number of files you can have in a project
+            projectSize: 0// total number of bytes for all files in this project
+    };
+    var currentProjectState = {
+        numOfFiles: 0 ,
+        projectSize: 0
+    };
+
+    ProjectLimiter.initMaxLimits =  function (projectState){
+        maxProjectState = projectState;
+        console.log(projectState);
+    };
+   function _checkLimits(){
+        if(currentProjectState.numOfFiles >= maxProjectState.fileCount || currentProjectState.projectSize >= maxProjectState.projectSize){
+            console.log("Maximum project size or file count is exceeded!\nCurrent number of files: "+currentProjectState.numOfFiles 
+            + "\nCurrent project size: " + currentProjectState.projectSize + "\nMaximum project file count: "
+            + maxProjectState.fileCount + "\nMaximum project size: " + maxProjectState.projectSize);
+        }
+    };
+
+    ProjectLimiter.newFile = function(){
+            currentProjectState.numOfFiles +=1;
```

Comment:
It's interesting that for some of the data you're keeping track of it as it happens, and for other data you're walking the filetree to figure things out.  You should probably pick one.  I think I'd do the latter, since you're already walking the entire filetree, and could easily extract the file count that way.